src/main/java/eu/dissco/core/translator/terms/DigitalObjectDirector.java

```diff
@@ -0,0 +1,595 @@
+            OdsTopicDiscipline.fromValue(termMapper.retrieveTerm(new TopicDiscipline(), data, dwc)))
+        .withDctermsLicense(termMapper.retrieveTerm(new License(), data, dwc))
+        .withOdsLivingOrPreserved(termMapper.retrieveTerm(new LivingOrPreserved(), data, dwc))
+        .withDwcPreparations(termMapper.retrieveTerm(new ObjectType(), data, dwc))
+        .withDwcCollectionId(termMapper.retrieveTerm(new PhysicalSpecimenCollection(), data, dwc))
+        .withDctermsModified(termMapper.retrieveTerm(new Modified(), data, dwc))
+        .withDwcInstitutionName(
+            rorComponent.getRoRId(minifyOrganisationId(ds.getDwcInstitutionId())))
+        .withDwcRecordedBy(termMapper.retrieveTerm(new RecordedBy(), data, dwc))
+        .withDwcBasisOfRecord(termMapper.retrieveTerm(new BasisOfRecord(), data, dwc))
+        .withDctermsAccessRights(termMapper.retrieveTerm(new AccessRights(), data, dwc))
+        .withDctermsRightsHolder(termMapper.retrieveTerm(new RightsHolder(), data, dwc))
+        .withDwcDatasetName(termMapper.retrieveTerm(new DatasetName(), data, dwc))
+        .withDwcDisposition(termMapper.retrieveTerm(new Disposition(), data, dwc));
+    return ds;
+  }
+
+  private List<eu.dissco.core.translator.schema.EntityRelationships> assembleDigitalSpecimenEntityRelationships(
+      DigitalSpecimen ds) {
+    var relationships = new ArrayList<EntityRelationships>();
+    relationships.add(new EntityRelationships().withEntityRelationshipType("hasOrganisationId")
+        .withObjectEntityIri(ds.getDwcInstitutionId()));
+    relationships.add(new EntityRelationships().withEntityRelationshipType("hasSourceSystemId")
+        .withObjectEntityIri(ds.getOdsSourceSystem()));
+    if (ds.getOdsPhysicalSpecimenIdType().equals(OdsPhysicalSpecimenIdType.RESOLVABLE)) {
+      relationships.add(
+          new EntityRelationships().withEntityRelationshipType("hasPhysicalIdentifier")
+              .withObjectEntityIri(ds.getOdsPhysicalSpecimenId()));
+    }
+    if (ds.getDctermsLicense() != null && ds.getDctermsLicense().startsWith("http")) {
+      relationships.add(new EntityRelationships().withEntityRelationshipType("hasLicense")
+          .withObjectEntityIri(ds.getDctermsLicense()));
+    }
+    if (ds.getCitations() != null) {
+      for (Citations citation : ds.getCitations()) {
+        if (citation.getReferenceIri() != null) {
+          relationships.add(new EntityRelationships().withEntityRelationshipType("hasReference")
+              .withObjectEntityIri(citation.getReferenceIri()));
+        }
+      }
+    }
+    return relationships;
+  }
+
+  private List<Identifiers> assembleIdentifiers(JsonNode data) {
+    var identifiers = new ArrayList<Identifiers>();
+    for (String identifierTerm : identifierTerms()) {
+      if (data.get(identifierTerm) != null) {
+        var identifier = new Identifiers()
+            .withIdentifierType(identifierTerm)
+            .withIdentifierValue(data.get(identifierTerm).asText());
```

Comment:
Oef I wouldn't know how to know if the id is persistent. When is the ID persistent... I could add it to the ID list. Some are clearly not dwca:ID but others could be.... tricky to determine if they are persistent. Maybe if they start with a handle or a http but still are http address persistent.

Reply:
https://naturalis.atlassian.net/browse/DD-725 Not sure if it will be possible for anything to determine if it is persistent. Persistency is a difficult beast but I added a US so we won't forget